app/controllers/main_routes/main_routes.py

```diff
@@ -47,30 +47,35 @@ def supervisorPortal():
     
     terms = LaborStatusForm.select(LaborStatusForm.termCode).distinct().order_by(LaborStatusForm.termCode.desc())
     allSupervisors = Supervisor.select()
+    supervisorFirstName = fn.COALESCE(Supervisor.preferred_name, Supervisor.legal_name)
+    studentFirstName = fn.COALESCE(Student.preferred_name, Student.legal_name)
     if currentUser.isLaborAdmin or currentUser.isFinancialAidAdmin or currentUser.isSaasAdmin:
         departments = Department.select().order_by(Department.DEPT_NAME.asc())
         departments = [department for department in departments]
-        supervisors = Supervisor.select().order_by(Supervisor.preferred_name, Supervisor.legal_name)
-        students = Student.select().order_by(Student.preferred_name, Student.legal_name)
+        supervisors = (Supervisor.select(Supervisor, supervisorFirstName.alias('first_name'))
+                                 .order_by(Supervisor.isActive.desc(), supervisorFirstName))
+        students = (Student.select(Student, studentFirstName.alias('first_name'))
+                           .order_by(studentFirstName))
```

Comment:
We don't ever use these coalesced names, do we? In your html you use the pre-existing FIRST_NAME property on the student and supervisor. We should just be able to drop the arguments to `select()`

Reply:
Fixed